extensions/amp-video-service/0.1/amp-video-service.js

```diff
@@ -76,14 +140,157 @@ export class VideoService {
 
   /**
    * @param {!AmpElement} unusedVideo
-   * @param {!../../../src/observable.Observable<boolean>} unusedObservable
+   * @param {!../../../src/observable.ObservableInterface<boolean>}
+   *   unusedObservable
    */
   delegateAutoplay(unusedVideo, unusedObservable) {
     warnUnimplemented('Autoplay delegation');
   }
 }
 
 
+/** @visibleForTesting */
+export class VideoEntry {
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!../../../src/video-interface.VideoInterface} video
+   * @param {!../../../src/observable.ObservableInterface<void>} tick
+   */
+  // Observables injected for testability.
+  constructor(ampdoc, video, tick) {
+
+    /** @private @const{!../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampdoc_ = ampdoc;
+
+    /** @private @const{!../../../src/video-interface.VideoInterface} */
+    this.video_ = video;
+
+    /** @private @const {!../../../src/observable.ObservableInterface<void>} */
+    this.tick_ = tick;
+
+    // These functions are bound (instead of being arrow functions) so that
+    // they reference the prototype of the class, which is only created once.
```

Comment:
Not sure what this means. Doesn't `this.onTick_ == VideoEntry.prototype.onTick_`?